sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/FaultInjectionRule.java

```diff
@@ -0,0 +1,173 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.models;
+
+import com.azure.cosmos.implementation.ImplementationBridgeHelpers;
+import com.azure.cosmos.implementation.faultinjection.model.IFaultInjectionRuleInternal;
+
+import java.net.URI;
+import java.time.Duration;
+import java.util.List;
+
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkNotNull;
+
+/***
+ * Fault injection rule.
+ */
+public class FaultInjectionRule {
+    private final FaultInjectionCondition condition;
+    private final IFaultInjectionResult result;
+    private final Duration duration;
+    private final Duration startDelay;
+    private final Integer hitLimit;
+    private final String id;
+    private boolean enabled;
+    private IFaultInjectionRuleInternal effectiveRule;
+
+    FaultInjectionRule(
+        String id,
+        Duration startDelay,
+        Duration duration,
+        Integer requestHitLimit,
+        boolean enabled,
+        FaultInjectionCondition condition,
+        IFaultInjectionResult result) {
+
+        checkNotNull(condition, "Argument 'condition' can not be null");
+        checkNotNull(result, "Argument 'result' can not be null");
+
+        this.id = id;
+        this.startDelay = startDelay;
+        this.duration = duration;
+        this.hitLimit = requestHitLimit;
+        this.enabled = enabled;
+        this.condition = condition;
+        this.result = result;
+    }
+
+    /***
+     * Flag to indicate whether the fault injection rule is enabled.
+     * By default, it is enabled.
+     *
+     * @param enabled flag to indicate whether the rule is enabled.
+     */
+    public void enabled(boolean enabled) {
```

Comment:
either enable and diable = or setEnabled?

Verb + thing like pattern.

Reply:
I removed the enabled(boolean) method, because once the rule is created, you can only disable it, if you want to enable it again, you will have to go through the CosmosFaultInjectionHelper.configureXXX flow.